src/libuvc/libuvc.cpp

```diff
@@ -402,42 +407,173 @@ namespace librealsense
+                    case RS2_OPTION_ENABLE_AUTO_WHITE_BALANCE:
+                        return UVC_PU_WHITE_BALANCE_TEMPERATURE_AUTO_CONTROL;
+                    case RS2_OPTION_POWER_LINE_FREQUENCY :
+                        return UVC_PU_POWER_LINE_FREQUENCY_CONTROL;
+                    case RS2_OPTION_AUTO_EXPOSURE_PRIORITY:
+                        unit = _input_terminal;
+                        return UVC_CT_AE_PRIORITY_CONTROL;
+                    default:
+                        throw linux_backend_exception(to_string() << "invalid option : " << option);
+                }
+            }
+
+            int32_t get_data_usb( uvc_req_code action, int control, int unit) const {
+                unsigned char buffer[4];
+
+                int status = libusb_control_transfer(_device_handle->usb_devh,
+                                                     UVC_REQ_TYPE_GET,
+                                                     action,
+                                                     control << 8,
+                                                     unit << 8 | (_interface),
+                                                     buffer,
+                                                     sizeof(int32_t), 0);
+
+                if (status < 0) throw std::runtime_error(
+                            to_string() << "libusb_control_transfer(...) returned "
+                                        << libusb_error_name(status));
+
+                if (status != sizeof(int32_t))
+                    throw std::runtime_error("insufficient data read from USB");
+
+                return DW_TO_INT(buffer);
+            }
+
+            void set_data_usb( uvc_req_code action, int control, int unit, int value) const {
+                unsigned char buffer[4];
+
+                INT_TO_DW(value, buffer);
+
+                int status = libusb_control_transfer(_device_handle->usb_devh,
+                                                     UVC_REQ_TYPE_SET,
+                                                     action,
+                                                     control << 8,
+                                                     unit << 8 | (_interface),
+                                                     buffer,
+                                                     sizeof(int32_t), 0);
+
+                if (status < 0) throw std::runtime_error(
+                            to_string() << "libusb_control_transfer(...) returned "
+                                        << libusb_error_name(status));
+
+                if (status != sizeof(int32_t))
+                    throw std::runtime_error("insufficient data writen to USB");
+            }
+
             bool get_pu(rs2_option opt, int32_t& value) const override
             {
-              // TODO : implement.
+                int unit;
+                int control = rs2_option_to_ctrl_selector(opt, unit);
```

Comment:
rs2_option_to_ctrl_selector(... initializes "unit" variable only for 3 specific values. In other cases it will remain uninitialized and passed on to get_data_usb